tracer/src/Datadog.Trace/Ci/ImpactedTestsModule.cs

```diff
@@ -0,0 +1,186 @@
+        {
+            Log.Warning("No test definition file found for {TestName}", tags.Name);
+            return Array.Empty<FileCoverageInfo>();
+        }
+
+        // Milestone 1 : Return only the test definition file
+        var file = new FileCoverageInfo(tags.SourceFile);
+
+        // Milestone 1.5 : Return the test definition lines
+        var prBase = BaseCommit;
+        if (prBase is not null)
+        {
+            var executedBitmap = FileBitmap.FromActiveRange((int)tags.SourceStart, (int)tags.SourceEnd);
+            file.ExecutedBitmap = executedBitmap.GetInternalArrayOrToArrayAndDispose();
+            Log.Debug<string, int, int>("TestCoverage for {TestFile}: {Start}..{End}", tags.SourceFile, (int)tags.SourceStart, (int)tags.SourceEnd);
+        }
+
+        return [file];
+    }
+
+    private static FileCoverageInfo[] GetModifiedFiles()
+    {
+        if (modifiedFiles is null)
+        {
+            lock (Log)
+            {
+                if (modifiedFiles is null)
+                {
+                    var workspacePath = CIEnvironmentValues.Instance.WorkspacePath ?? string.Empty;
+                    var prBase = BaseCommit;
+                    var commit = CurrentCommit;
+                    if (prBase is { Length: > 0 })
+                    {
+                        Log.Debug("PR detected. Retrieving diff lines from Git CLI for {Path} {BaseCommit}...{CurrentCommit}", workspacePath, prBase, commit);
+                        // Milestone 1.5 : Retrieve diff files and lines from Git Diff CLI
+                        try
+                        {
+                            modifiedFiles = AsyncUtil.RunSync(() => GitCommandHelper.GetGitDiffFilesAndLinesAsync(workspacePath, prBase, commit));
+                        }
+                        catch (Exception ex)
+                        {
+                            Log.Debug(ex, "Git command failed.");
+                        }
+                    }
+                    else
+                    {
+                        Log.Debug("No PR detected.");
+                    }
+
+                    if (modifiedFiles is null)
+                    {
+                        // Milestone 1 : Retrieve diff files from Backend
+                        modifiedFiles = GetDiffFilesFromBackend();
+                    }
+                }
+            }
+        }
+
+        return modifiedFiles;
+    }
+
+    private static FileCoverageInfo[] GetDiffFilesFromBackend()
+    {
+        Log.Debug("Retrieving diff files from backend...");
+
+        if (CIVisibility.ImpactedTestsDetectionResponse is { } response && response.Files is { Length: > 0 } files)
+        {
+            var res = new List<FileCoverageInfo>();
+            foreach (var file in files)
+            {
+                res.Add(new FileCoverageInfo(file));
+            }
+
+            return res.ToArray();
+        }
```

Comment:
We have the Length, we can create the FileCoverageInfo[] array and avoid creating the list, resizing the list and the calling `ToArray()`

Reply:
Agree